ampcrowd/basecrowd/models.py

```diff
@@ -150,3 +171,43 @@ def add_model_rels(self):
         # responses pertain to a task
         self.add_rel(self.response_model, self.task_model, models.ForeignKey,
                      'task', 'responses')
+
+
+class TemplateResource(models.Model):
+    name = models.CharField(max_length=200)
+    content = models.TextField()
+    direct_dependencies = models.ManyToManyField('self', symmetrical=False, related_name="upstream_dependencies")
+    direct_requirements = models.ManyToManyField('self', symmetrical=False, related_name="upstream_requirements")
+
+    @property
+    def dependencies(self):
+        query_set = list(self.direct_dependencies.all())
+        all_dependencies = set(query_set)
+        current_dependencies = list(query_set)
+        while True:
+            prior_size = len(all_dependencies)
+            new_dependencies = set()
+            for dep in current_dependencies:
+                deps = set(dep.direct_dependencies.all())
+                new_dependencies = new_dependencies.union(deps)
+                all_dependencies = all_dependencies.union(deps)
+            if prior_size == len(all_dependencies):
+                break
+            else:
+                current_dependencies = list(new_dependencies)
+        return all_dependencies
+
```

Comment:
This looks pretty painfully inefficient in terms of DB queries. I'll put a little thought into it, but there should be an easier way to do this.